src/Application/Components/MapCanvas/MapCanvas.jsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { fabric } from 'fabric-webpack';
+
+/**
+ * No Pin functionality yet.
+ */
+const MapCanvas = React.createClass({
+  propTypes: {
+    pins: React.PropTypes.object,
```

Comment:
Word, just checked it all out. It's in the works, seems like immutable is pretty standard stuff for react.